app/src/androidTest/java/com/github/se/project/ui/profile/EditTutorProfileTest.kt

```diff
@@ -0,0 +1,172 @@
+    mockViewModel = Mockito.spy(ListProfilesViewModel(mockProfilesRepository))
+    doReturn(mutableStateFlow as StateFlow<Profile?>).`when`(mockViewModel).currentProfile
+  }
+
+  @Test
+  fun editTutorProfileScreen_rendersCorrectly() {
+    (mockViewModel.currentProfile as MutableStateFlow).value =
+        Profile(
+            uid = "1",
+            googleUid = "googleUid",
+            firstName = "First",
+            lastName = "Last",
+            phoneNumber = "1234567890",
+            role = Role.STUDENT,
+            section = Section.GM,
+            academicLevel = AcademicLevel.MA2,
+            languages = listOf(Language.ENGLISH),
+            subjects = listOf(Subject.ALGEBRA),
+            schedule = listOf())
+    // Set the screen in the test environment
+    composeTestRule.setContent { EditTutorProfile(mockNavigationActions, mockViewModel) }
+
+    // Assert all expected UI components are visible
+    // composeTestRule.onNodeWithTag("editTutorProfileCloseButton").assertExists()
+    composeTestRule.onNodeWithTag("editTutorWelcomeText").assertIsDisplayed()
+    composeTestRule.onNodeWithTag("editTutorProfileInstructionText").assertIsDisplayed()
+    composeTestRule.onNodeWithTag("editTutorProfileSectionDropdown").assertIsDisplayed()
+    composeTestRule.onNodeWithTag("editTutorProfileAcademicLevelDropdown").assertIsDisplayed()
+    composeTestRule.onNodeWithTag("editTutorProfilePriceText").assertIsDisplayed()
+    composeTestRule.onNodeWithTag("editTutorProfileConfirmButton").assertIsDisplayed()
+  }
+
+  @Test
+  fun editTutorProfileUpdatesCorrectly() {
+    (mockViewModel.currentProfile as MutableStateFlow).value =
+        Profile(
+            uid = "12345",
+            googleUid = "67890",
+            firstName = "John",
+            lastName = "Doe",
+            phoneNumber = "1234567890",
+            role = Role.TUTOR,
+            section = Section.IN,
+            academicLevel = AcademicLevel.MA2,
+            languages = listOf(Language.ENGLISH),
+            subjects = listOf(Subject.ALGEBRA),
+            schedule = List(7) { List(12) { 0 } })
+    // Set the screen in the test environment
+    composeTestRule.setContent { EditTutorProfile(mockNavigationActions, mockViewModel) }
+
+    composeTestRule.onNodeWithTag("editTutorProfileAcademicLevelDropdown").performClick()
+    composeTestRule
+        .onNodeWithTag("editTutorProfileAcademicLevelDropdownItem-MA4")
+        .assertIsDisplayed()
+        .performClick()
+
+    composeTestRule.onNodeWithTag("editTutorProfileSectionDropdown").performClick()
+    composeTestRule
+        .onNodeWithTag("editTutorProfileSectionDropdownItem-GM")
+        .assertIsDisplayed()
+        .performClick()
+
+    composeTestRule.onNodeWithTag("editTutorProfileConfirmButton").performClick()
+
+    assertEquals(AcademicLevel.MA4, mockViewModel.currentProfile.value?.academicLevel)
+    assertEquals(Section.GM, mockViewModel.currentProfile.value?.section)
```

Comment:
Could be useful to have an equivalent test that also edits languages/subjects/name, but otherwise good job